T77 main

dlahtine opened the pull request:
Must merge flask_server (https://github.com/zawata/S19-Team2/pull/27) and t80 (https://github.com/zawata/S19-Team2/pull/24) first.

1. in config, create config.json with the following contents:
`{
  "main_subject": -39,
  "subject_name": "LMAP",
  "kernels": [
    "LMAP_FullTrajectory.bsp"
  ]
}`

2. in config/kernels, add LMAP_FullTrajectory.bsp from the team drive
3. build and run
4. make a get request to the /api/main_object, you should get the following result
`{
"id": -39,
"name": "LMAP"
}`
5. make a get request to the /api/all_objects?time=600000000 you should get the same as before, but with position and velocity information.

dlahtine committed:
Merge branch 't40_frame_data' of https://github.com/zawata/S19-Team2 into flask_server

dlahtine committed:
Merge branch 'flask_server' of https://github.com/zawata/S19-Team2 into flask_server

dlahtine committed:
Rearranged items

dlahtine committed:
Rearranged items

dlahtine committed:
Should load spacecraft from config properly.

dlahtine committed:
Should load spacecraft from config properly.

dlahtine committed:
Merge branch 'dev' of https://github.com/zawata/S19-Team2 into flask_server

dlahtine committed:
Should load spacecraft from config properly.

dlahtine committed:
Should load spacecraft from config properly.

dlahtine committed:
Now calls add_kernel() on main file

dlahtine committed:
Now calls add_kernel() on main file

dlahtine committed:
reorganized config directory

dlahtine committed:
reorganized config directory

dlahtine committed:
get_objects is now called for all kernels

dlahtine committed:
get_objects is now called for all kernels

dlahtine committed:
main_file is reset after get_objects call.

dlahtine committed:
main_file is reset after get_objects call.

zawata committed:
add id_to_str function, move python declaration to module

zawata committed:
add id_to_str function, move python declaration to module

dlahtine committed:
get_objects now properly resets the main file for the kernel

dlahtine committed:
get_objects now properly resets the main file for the kernel

dlahtine committed:
Merge branches 'dev' and 'flask_server' of https://github.com/zawata/S19-Team2 into flask_server

# Conflicts:
#	FlaskServer.py

dlahtine committed:
changed the way kerneles were configured.

dlahtine committed:
changed the way kerneles were configured.

dlahtine committed:
added main_subject back in

dlahtine committed:
added main_subject back in

dlahtine committed:
added /api/main_id endpoint for retrieving id number of spacecraft.

dlahtine committed:
added /api/main_id endpoint for retrieving id number of spacecraft.

dlahtine committed:
Changed config file structure, all_objects now returns names of objects
as well as frame data.

dlahtine committed:
Changed config file structure, all_objects now returns names of objects
as well as frame data.

dlahtine committed:
Merge branches 't77_main_id' and 't80_id_str' of https://github.com/zawata/S19-Team2 into t77_main_id

dlahtine committed:
removed prints

dlahtine committed:
removed prints

dlahtine committed:
changed all_objects endpoint's time param to be optional.
Will only get frame data with time param present

dlahtine committed:
changed all_objects endpoint's time param to be optional.
Will only get frame data with time param present

dlahtine committed:
Merge branches 'flask_server' and 't77_main_id' of https://github.com/zawata/S19-Team2 into t77_main_id

# Conflicts:
#	FlaskServer.py

dlahtine committed:
fixed equality check bug

dlahtine committed:
fixed equality check bug

dlahtine committed:
Merge branches 'flask_server' and 't77_main_id' of https://github.com/zawata/S19-Team2 into t77_main_id

dlahtine committed:
removed comment

dlahtine committed:
removed comment

dlahtine committed:
Merge branch 't77_main_id' of https://github.com/zawata/S19-Team2 into t77_main_id

# Conflicts:
#	FlaskServer.py
#	spyce/spyce_main.cpp

dlahtine committed:
Merge branches 'dev' and 't77_main_id' of https://github.com/zawata/S19-Team2 into t77_main_id

# Conflicts:
#	FlaskServer.py
#	spyce/spyce_main.cpp

trevorforrey committed:
Fix: Spyce import

trevorforrey reviewed (commented):
`/main_object` and `/getobjects?time=XXXX` works.

However, when testing `/getobjects` I got the following error:

> builtins.UnboundLocalError
> UnboundLocalError: local variable 'frame_as_dict' referenced before assignment

dlahtine committed:
Merge branches 'dev' and 't77_main_id' of https://github.com/zawata/S19-Team2 into t77_main_id

# Conflicts:
#	FlaskServer.py
#	spyce/spyce_main.cpp

dlahtine committed:
Merge branch 't77_main_id' of https://github.com/zawata/S19-Team2 into t77_main_id

dlahtine committed:
changed import, fixed all_objects endpoint

dlahtine committed:
now successfully handles all_objects with leapseconds loaded.

dlahtine committed:
added error handling for objects with no data for given timestamp

dlahtine commented:
> `/main_object` and `/getobjects?time=XXXX` works.
> 
> However, when testing `/getobjects` I got the following error:
> 
> > builtins.UnboundLocalError
> > UnboundLocalError: local variable 'frame_as_dict' referenced before assignment

Found the problem; solved now.

trevorforrey committed:
Refix: importing spyce

Please do not revert this. It needs to be like this for the docker build to run correctly

trevorforrey reviewed (approved):
Tested just now and all the routes added pass! Great job!
